main.cpp

```diff
@@ -2,29 +2,42 @@
 #include "ElectricCar.hpp"
 #include "HybridCar.hpp"
 #include <iostream>
+#include <cassert>
 
 int main()
 {
     Car* car = nullptr;
     std::cout << std::endl << "OPEL" << std::endl;
     PetrolCar opel(new PetrolEngine(120, 1800, 6));
     car = &opel;
+    car->changeGear(1);
+    assert(car->getCurrentGear() == "1");
     car->accelerate(50);
+    assert(car->getCurrentSpeed() == 50);
     car->brake();
-    car->accelerate(-900);
+    assert(car->getCurrentSpeed() == 0);
+    car->changeGear(0);
+    assert(car->getCurrentGear() == "0");
+    car->changeGear("R");
+    assert(car->getCurrentGear() == "R");
+    car->accelerate(-9);
+    assert(car->getCurrentSpeed() == -9);
```

Comment:
jesli juz masz gtesta to moze lepiej byloby napisac test casy?

Reply:
To tylko sugestia, w tym zadaniu to nie jest wymagane.